library/core/src/ffi/c_str.rs

```diff
@@ -882,7 +882,32 @@ mod verify {
             assert!(c_str.is_safe());
         }
     }
-  
+
+    // pub const fn from_bytes_with_nul(bytes: &[u8]) -> Result<&Self, FromBytesWithNulError>
+    #[kani::proof]
+    #[kani::unwind(17)]  
+    fn check_from_bytes_with_nul() {
+        const MAX_SIZE: usize = 16;
+        let mut string: [u8; MAX_SIZE] = kani::any();
+        let null_pos: usize = kani::any_where(|x| *x < MAX_SIZE);
```

Comment:
Since you aren't doing the loop anymore, can you go back to 32?

Reply:
I tried doing that, but it fails due to insufficient memory with and without loops(Im running on a 32GB Ram M2 machine), Hence limited it to 16

Reply:
I tried doing that, but it fails due to insufficient memory, with and without loops(Im running on a 32GB Ram M2 machine), Hence limited it to 16.

I believe this is due to the use of memchar in the original function, which could expand to larger state spaces with more bits, and also the function calls the `from_bytes_with_nul_unchecked` function internally. which contains a loop that iterates from the end of the slice backwards to ensure no interior null bytes exist.